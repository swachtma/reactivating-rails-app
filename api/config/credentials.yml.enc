NruSqAfDHjCumPIyJn6xsY9LOZuE3ZzX2o/1QTdVkugVslYlIp4h/gmrnNPKDUKYZ8vGEGHPJ5EwlOSClaonGjgUB4yUG4tz8I0T++XLbiEU8qhxATlTf+s9JNfusXdgIB2ZFO4ABSDKhxhzPLbHYJVvLghx3WmwMMpO+GBgzlp94DLCX2IR38jeHEydFPy73oteodKVbTBp0g91v8DhpnZUTWHYb0JdqS+otrdrS1mar9MlcMnTpb8chvLisrrvE0SR/QUagAbd5UD/d02NczFYmAdKsyztbryVyakhgosZhCJFY3KNwxayxDC3sF5xFQHgli8qJNXNAhzaYDVf+n7D9k6g9BHTV5YRXdICgYORVWo+a9EhjBTmVwQsJN+IrdZaqxQVoa2DMEA3CVjOlp1wtLNwHbHArmId--8+ArDlMZvB3FEYTs--m4O75Mk62PJos0rR8zw91Q==