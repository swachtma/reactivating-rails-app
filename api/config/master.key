dc4eeb306d00d2a5e6993b94e449787e