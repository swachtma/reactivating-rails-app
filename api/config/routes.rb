Rails.application.routes.draw do
  scope :api do
    get 'github/', to: "authentication#github", format: false
    get 'hydrate_user/', to: "authentication#show"
    get 'nodes/', to: "nodes#index"
    get 'chapters/', to: "chapters#index"
    resources :bookmarks, only: [:index, :create]
  end
end