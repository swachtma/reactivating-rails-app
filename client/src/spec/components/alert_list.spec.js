/* global expect, jest */
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import config, { expectXInY } from '../../test_helper';
import AlertList from '../../components/alert_list';

let test_props = {
  scope: "application",
  alerts: [
    {
      message: "Sign out successful",
      level: "warning",
      scope: "application",
      timestamp: 1529324313949,
    },
    {
      message: "Cylons are among us",
      level: "danger",
      scope: "battlestar",
      timestamp: 1529324313939,
    }
  ]
};

describe("<AlertList /> renders",()=>{
  let w = shallow(<AlertList {...test_props} />);
  it("renders a list for the scope", () => expectXInY(".alert_list_"+test_props.scope,w));
  it("doesn't render alerts out of scope", ()=> { expect(w.find(".alert").length).toBe(1)});
  it("according to snapshot", () => expect(w).toMatchSnapshot());
});
