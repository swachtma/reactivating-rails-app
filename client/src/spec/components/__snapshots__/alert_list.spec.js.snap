// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertList /> renders according to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AlertList
    alerts={
      Array [
        Object {
          "level": "warning",
          "message": "Sign out successful",
          "scope": "application",
          "timestamp": 1529324313949,
        },
        Object {
          "level": "danger",
          "message": "Cylons are among us",
          "scope": "battlestar",
          "timestamp": 1529324313939,
        },
      ]
    }
    scope="application"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="alert alert-warning"
          style={
            Object {
              "backgroundColor": "#f2dede",
              "border": "1px solid transparent",
              "borderColor": "#ebccd1",
              "borderRadius": "4px",
              "color": "#a94442",
              "marginBottom": "20px",
              "padding": "15px",
            }
          }
        >
          Sign out successful
        </div>,
        false,
      ],
      "className": "alert_list_application",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1529324313949",
        "nodeType": "host",
        "props": Object {
          "children": "Sign out successful",
          "className": "alert alert-warning",
          "style": Object {
            "backgroundColor": "#f2dede",
            "border": "1px solid transparent",
            "borderColor": "#ebccd1",
            "borderRadius": "4px",
            "color": "#a94442",
            "marginBottom": "20px",
            "padding": "15px",
          },
        },
        "ref": null,
        "rendered": "Sign out successful",
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="alert alert-warning"
            style={
              Object {
                "backgroundColor": "#f2dede",
                "border": "1px solid transparent",
                "borderColor": "#ebccd1",
                "borderRadius": "4px",
                "color": "#a94442",
                "marginBottom": "20px",
                "padding": "15px",
              }
            }
          >
            Sign out successful
          </div>,
          false,
        ],
        "className": "alert_list_application",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1529324313949",
          "nodeType": "host",
          "props": Object {
            "children": "Sign out successful",
            "className": "alert alert-warning",
            "style": Object {
              "backgroundColor": "#f2dede",
              "border": "1px solid transparent",
              "borderColor": "#ebccd1",
              "borderRadius": "4px",
              "color": "#a94442",
              "marginBottom": "20px",
              "padding": "15px",
            },
          },
          "ref": null,
          "rendered": "Sign out successful",
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
